src/main/java/software/amazon/cloudformation/resource/ResourceTypeSchema.java

```diff
@@ -153,6 +149,57 @@ public ResourceTypeSchema(Schema schema) {
             });
             return null;
         });
+
+        this.unprocessedProperties.computeIfPresent("tagging", (k, v) -> {
+            ((Map<?, ?>) v).forEach((key, value) -> {
+                if (!key.equals("tagProperty") && !key.equals("tagPropertyType")) {
+                    this.tagging.put(key.toString(), Boolean.valueOf(value.toString()));
+                } else if (key.equals("tagProperty")) {
+                    this.tagging.put(key.toString(), new JSONPointer(value.toString()));
+                } else {
+                    this.tagging.put(key.toString(), value.toString());
+                }
+            });
+
+            if (this.tagging.containsKey("taggable") && this.tagging.containsKey("tagOnCreate")
+                && this.tagging.get("taggable").equals(false) && this.tagging.get("tagOnCreate").equals(true)) {
+                throw new ValidationException("Invalid tagOnCreate value since taggable is marked false", "tagging",
+                                              "#/tagging/tagOnCreate");
+            } else if (this.tagging.containsKey("taggable") && this.tagging.containsKey("tagUpdatable") &&
+                this.tagging.get("taggable").equals(false) && this.tagging.get("tagUpdatable").equals(true)) {
+                throw new ValidationException("Invalid tagUpdatable value since taggable is marked false", "tagging",
+                                              "#/tagging/tagUpdatable");
+            } else if (this.tagging.containsKey("taggable") && this.tagging.containsKey("CFNSystemTags") &&
```

Comment:
As those are validation check,  I would suggest to move into a separated method, and do not need `else if` for each validation.

Reply:
sure have revised